### ISSUE TITLE
subscription in component vs in service

### ISSUE THREAD

**Issue by vlapenkov** [ngrx/platform#1603]
I'm newbee in Redux (ngrx). I have simple app:  FilterComponent, ListComponent and one AppStore. 
```
export interface AppState {    
    login:ILoginData,
    carItems:ICarItem[],  //resultset
    filter:ICarItem,  // filter
  }
```
When user change one of filters on form, FilterComponent notifies store that filters changed. 
`this.store.dispatch(new SetFilterAction(  newFilter));`
 Than i need to make http get request to get resultset and make dispatch: `this.store.dispatch(new LoadCarItemsAction(list))`. 
The question is : should i place second step in ListComponent or create service and subscribe in the service, and this service already makes dispatch.
In other words who is responsible for the following logic:
`return  this.store.select(state => state.filter).pipe(
           map (source=> 
            {
            let productId=source.product ? source.product.productId:'';
            let brand = source.brand||'';        
            return({productId ,brand})            
            }),
        switchMap(value => this.http.get(`${this._baseUrl}api/caritems`,{params:value}))`

Thanks in advance!

**Comment by timdeschryver** [comment]
Hi, this is where most of the NgRx users are using [@ngrx/effects](https://ngrx.io/guide/effects) for. An effect handles all of the side effects within an application using NgRx, and can be triggered when you dispatch an action.

In your case the effect will call the service when it receives the `SetFilterAction` (or `LoadCarItemsAction`?) action. When the service returns data, the effect will dispatch a `LoadCarItemsSuccessAction` containing all of the fetched cars. You will handle this dispatched action inside a reducer, modifying the state.

In your component you will have to select the car state from the store, this will be an observable. Inside your component's template you can use the AsyncPipe to render the cars.


Maybe this small diagram could be of help:
![ngrx-flow](https://user-images.githubusercontent.com/28659384/53826990-45bbb400-3f7a-11e9-983d-8c99d92c8ec9.gif)
